src/plugins/xref/schema.js

```diff
@@ -17,7 +17,8 @@ export default {
             // TODO: better data validation
             data: {
                 document: value => value == null || typeof value === 'string',
-                target: Boolean,
+                target: t => t === null || typeof t === 'string',
+                declension: d => d === null || d === undefined || typeof d === 'string',
```

Comment:
1. `target` must be defined, so this check is wrong
2. rather than testing for `null` and `undefined` separately, check just for `d == null`

Reply:
Not all xrefs will have `data.declension` so `undefined` is needes as well as `null`

Reply:
`target: typeof t === 'string'` will be ok then?

Reply:
1. Note that I proposed `d == null`, with only two equals signs. This is the same as checking with three equal signs for `d === null || d === undefined`

2. Technically in Slate keys need not be strings (or at least I think so), that's why I was only checking for their presence and not type

Reply:
`d === null || d === undefined` - this is exactly why I've like to separate those checks, but it's your repo so I'll adjust

Reply:
In general I agree that it's better to use explicit checks, but here we have so many checks that it just becomes hard to read.